### PR TITLE
Javascript javascript1 week2

### PR DESCRIPTION
**Pull request by tanya-newStart**

### CODE REVIEW

File: javascript/javascript1/week2/boarding_pass.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+fullname1 = getFullname("Benjamin", "Hughes", true); // returns "Benjamin Hughes"
+fullname2 = getFullname("Tanya", "Dewland", true, true); //need another input of type "checkbox" for gender in html
+fullname3 = getFullname("Quick", false, true); //returns an error
+console.log(fullname1);
+console.log(fullname2);
+console.log(fullname3);
+
+getFullname("Benjamin", "Hughes", true); // returns "Lord Benjamin Hughes"`
+console.log(getFullname("Benjamin", "Hughes", false)); // returns "Benjamin Hughes"
+console.log(getFullname("", "")); //returns an error
```
Comment (Rassibassi) [inline javascript/javascript1/week2/boarding_pass.js:31]: Good job! Nice on throwing errors, probably somewhat overkill for this exercise :)

Get used to use string template literals more: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

File: javascript/javascript1/week2/candy_helper.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    amountToSpend = budget;
+    return `You can buy more, so please do! You have ${amountToSpend.toFixed(
+      2
+    )} to spend.`;
+  } else {
+    return `You are buying candy that costs ${price}.You have only ${amountToSpend.toFixed(
+      2
+    )} left. Enough candy for you?!`;
+  }
+}
```
Comment (Rassibassi) [inline javascript/javascript1/week2/candy_helper.js:25]: Good job!

This is a limited and self containers example, so it does not matter too much, but try in the future to not have your function return the log messages. try to make your functions do the "logic" and then have the logging outside of the function. Along the same lines, all the `toFixed` is for having "nice" output logs at the end. Could you live with only a `toFixed` at the very end after the logic and outside of the function? You can google "Separation of concerns", which is a concept that one wants to separate for example logic from rendering/outputting.

File: javascript/javascript1/week2/candy_helper.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+//USING FOR LOOP HERE:
+//function canBuyMoreCandy() {
+//  for (let i = 0; i < boughtCandyPrices.length; i++) {
+//    amountToSpend -= boughtCandyPrices[i];
+//  }
+// return amountToSpend > 0;
+//}
```
Comment (Rassibassi) [inline javascript/javascript1/week2/candy_helper.js:32]: Nice for loop! You could also count up like in your while loop below. Then the `amountToSpend` could even be `const`

File: javascript/javascript1/week2/date.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    "Wednesday",
+    "Thursday",
+    "Friday",
+    "Saturday",
+  ];
+  const day_index = (weekday + num) % 7;
+  return week[day_index];
+}
+console.log(getEventWeekday(1));
+console.log(getEventWeekday(2));
```
Comment (Rassibassi) [inline javascript/javascript1/week2/date.js:17]: Good work

File: javascript/javascript1/week2/student_manager.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    class07Students.push(studentName);
+    console.log("Welcome aboard, Your Majesty!");
+  } else if (class07Students.length < 6) {
+    class07Students.push(studentName);
+    console.log(`${studentName} has been added to the class`);
+  } else {
+    console.log("Class is full.");
+  }
+}
+
```
Comment (Rassibassi) [inline javascript/javascript1/week2/student_manager.js:22]: good work on first checking the negatives and then the positives :)

File: javascript/javascript1/week2/weather_wear.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  } else if (temp >= 12) {
+    return clothes[1];
+  } else if (temp >= 5) {
+    return clothes[2];
+  } else if (temp >= -5) {
+    return clothes[3];
+  } else {
+    return "Stay home! It's freezing!";
+  }
+}
```
Comment (Rassibassi) [inline javascript/javascript1/week2/weather_wear.js:19]: Nice! Can you think of a nested array and for-looping over temps and clothes synchronously, with a single if statement inside the for-loop?

**Review by Rassibassi — COMMENTED** [review]
Good job, comments inline!

File: javascript/javascript1/week2/candy_helper.js (hunk 2 above)
Comment (tanya-newStart) [reply javascript/javascript1/week2/candy_helper.js:25]: Yes! That makes sense!

File: javascript/javascript1/week2/weather_wear.js (hunk 6 above)
Comment (tanya-newStart) [reply javascript/javascript1/week2/weather_wear.js:19]: Yes! Sweet!

File: javascript/javascript1/week2/candy_helper.js (hunk 3 above)
Comment (tanya-newStart) [reply javascript/javascript1/week2/candy_helper.js]: I need to figure out a way to cut the `budget` or `amountToSpend` variables

**Pull request merged**